### ISSUE TITLE
Unable to test .tar.bz2 packages that use the test:source_files section

### ISSUE THREAD

**Issue by kuepe-sl** [conda/conda-build#2517]
I'm trying to build and test a package. The package includes test files from the original source tree. The test section of my meta.yaml file looks like this
```yaml
test:
  requires:
    - cmake
  source_files:
    - testdata
```

However, when I try to test the package with `conda-build --test ./example.tar.bz2`, I get the following error:  
`('Did not find any source_files for test with pattern %s', 'testdata')`  
The stack trace is pointing to conda_build/create_test.py, line 52, in create_files.

I can confirm that the required files are in the package, at `info/test/testdata`.
But it apparently doesn't extract the test files from the package to the working directory and/or fails to find them.

(It works if I test the conda recipe directory, but that is to be expected. But I don't think it's a good idea to use this as a workaround.)

**Comment by msarahan** [comment]
What conda build version?

**Comment by kuepe-sl** [comment]
I'm using conda-build 3.0.30 right now.

I just installed some previous versions and it broke between 3.0.28 and 3.0.29.

**Comment by msarahan** [comment]
Thanks for helping to narrow it down.  I'll fix it for the next bugfix release.

**Comment by kuepe-sl** [comment]
Are there any news on this?  
It doesn't seem to be fixed with CB 3.0.31 and I'm recently running into issues with the recipe tests.

**Comment by kuepe-sl** [comment]
I have good news and bad news.

Good news: The tests now work properly. (Packages built with conda-build 3.0.31 are tested successfully.)

Bad news: Now I get the error from above when *building* the package. It happens during the packaging process. (The last message I get before the exception occours is `Fixing permissions`.)

**Comment by mingwandroid** [comment]
Please show the exception?!

**Comment by kuepe-sl** [comment]
It's the exactly same error as mentioned in the first post (same exception, same location, etc), so I assumed the exact stack trace would be redundant, as the source of the issue was already tracked down. But for reference:
```
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 379, in execute
    verify=args.verify)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/api.py", line 187, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 1833, in build_tree
    notest=notest,
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 1285, in build
    built_package = bundlers[output_d.get('type', 'conda')](output_d, m, env)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 762, in bundle_conda
    output['checksums'] = create_info_files(metadata, files, prefix=metadata.config.host_prefix)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 526, in create_info_files
    copy_test_source_files(m)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 285, in copy_test_source_files
    create_all_test_files(m, test_dest)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/create_test.py", line 256, in create_all_test_files
    files = create_files(m, test_dir)
  File "/home/kuepe/miniconda3/lib/python3.5/site-packages/conda_build/create_test.py", line 50, in create_files
    raise RuntimeError("Did not find any source_files for test with pattern %s", pattern)
RuntimeError: ('Did not find any source_files for test with pattern %s', 'test')
```

I'm building with `conda-build conda-recipe/ --no-test`. The folder "conda-recipe" contains the meta.yaml file.

**Comment by msarahan** [comment]
This should be fixed with conda-build 3.1.0, released yesterday.

**Comment by kuepe-sl** [comment]
I did some additional research and it turned out to be a different issue that occurs only under very specific conditions (that my project happened to fullfill). see #2550
